### PR TITLE
undo subscribe for addying cart stuff and shows price and total count…

### PR DESCRIPTION
**Pull request by vikas-cis**
… of item added in the cart

#874

### CODE REVIEW

File: src/app/core-ui/main/main-view.component.ts [hunk 1]
```diff
@@ -32,7 +34,7 @@ export class MainViewComponent implements OnInit, OnDestroy {
 
   title: string = '';
   testnet: boolean = false;
-
+  cartItemCount: number = 0;
```
Comment (kewde) [inline src/app/core-ui/main/main-view.component.ts]: rm

File: src/app/core-ui/main/main-view.component.html [hunk 2]
```diff
@@ -238,15 +238,14 @@
         {{ title }}
       </div>
     </div><!-- .left -->
-
     <div class="right" fxFlex="1 1 60%">
       <!-- FIXME: show only when stuff is in cart -->
-      <div class="cart-info TO_DO" routerLinkActive="active"
+      <div class="cart-info" routerLinkActive="active"
```
Comment (kewde) [inline src/app/core-ui/main/main-view.component.html]: Can we move this code into a separate component?
like the status bar basically..

https://github.com/vikas-cis/partgui/tree/ab00a2e2fcd5b2c2909930dcedb6b5b497c25fe9/src/app/core-ui/main

File: src/app/core-ui/main/main-view.component.ts [hunk 3]
```diff
@@ -118,6 +120,15 @@ export class MainViewComponent implements OnInit, OnDestroy {
     /* check if testnet -> block explorer url */
     this._rpcState.observe('getblockchaininfo', 'chain').take(1)
       .subscribe(chain => this.testnet = chain === 'test');
+
```
Comment (kewde) [inline src/app/core-ui/main/main-view.component.ts]: move this into it too obv

File: src/app/core-ui/main/main-view.component.ts [hunk 4]
```diff
@@ -118,6 +120,15 @@ export class MainViewComponent implements OnInit, OnDestroy {
...
+    // Obtain total cart items
+    // @TODO move to market state service ?
+    this._marketState.observe('cart')
```
Comment (kewde) [inline src/app/core-ui/main/main-view.component.ts]: move this into it too obv

File: src/app/core-ui/main/main-view.component.html (hunk 2 above)
Comment (kewde) [reply src/app/core-ui/main/main-view.component.html]: also what's the routerLinkActive doing here, curious..

Comment (vikas-cis) [reply src/app/core-ui/main/main-view.component.html]: lol I think its doing nothing we can remove it 👍

File: src/app/core/market/api/cart/cart.service.ts [hunk 5]
```diff
@@ -15,13 +16,17 @@ export class CartService {
...
   ) { }
 
   addItem(listingItemId: number) {
     this.log.d(`Adding listingItemId=${listingItemId} to cart with id=1`);
-    this.market.call('cartitem', ['add', 1, listingItemId]).take(1).do(
-      data => this.snackbar.open(data),
+    this.market.call('cartitem', ['add', 1, listingItemId]).take(1).subscribe(
+      data => {
+        this.snackbar.open(data)
+        this.marketState.registerStateCall('cart', null, ['get', 1])
```
Comment (kewde) [inline src/app/core/market/api/cart/cart.service.ts]: Can we move this into the cart.service?

Comment (kewde) [inline src/app/core/market/api/cart/cart.service.ts]: Can we move this into the cart.service?

Comment (kewde) [reply src/app/core/market/api/cart/cart.service.ts]: Should update: on addition, on delete, on clear

Comment (vikas-cis) [reply src/app/core/market/api/cart/cart.service.ts]: lol it seems already in cart service :P

**Pull request merged**